synapse/lib/cache.py

```diff
@@ -95,3 +96,45 @@ def get(self, key):
     def clear(self):
         self.fifo.clear()
         self.cache.clear()
+
+class TagGlobs:
+    '''
+    An object that manages multiple tag globs and values for caching.
+    '''
+    def __init__(self):
+        self.globs = []
+        self.cache = FixedCache(self._getGlobMatches)
+
+    def add(self, name, valu, base=None):
+
+        self.cache.clear()
+
+        parts = []
+
+        for part in name.split('.'):
+            part = part.replace('*', '([^.]+)')
```

Comment:
I'm not super confident that `part` never contains any valid regex chars (other than *). Suggest using `re.escape(part)` and making the first parm to `replace` be `\*`.